tianshou/data/utils/converter.py

```diff
@@ -80,3 +82,58 @@ def to_torch_as(
     """
     assert isinstance(y, torch.Tensor)
     return to_torch(x, dtype=y.dtype, device=y.device)
+
+
+def to_hdf5(x: Any, y: h5py.Group) -> None:
```

Comment:
`Any` is not convertible to hdf5 format. You should be way more precise than that. Eventually:
`Hdf5ConvertibleType =  Dict[str, Union[np.ndarray, torch.Tensor, 'Hdf5ConvertibleType']]`, or if you don't support nested dictionary: `Hdf5ConvertibleType =  Dict[str, Union[np.ndarray, torch.Tensor]]`.

Reply:
Thank you for your feedback. I do support nested dictionaries. It is a bit difficult to say precisely what can be saved this way as the function falls back to using pickle. So it would actually be something like Dict[int, float, Batch, np.ndarray, torch.Tensor, 'Hdf5ConvertibleType', 'anything that can be pickled'].
I am not sure how to best express that.

Reply:
Hum OK I see the issue. There is no type associated with 'pickle-able'. Indeed, any `object` is `pickle-able`, so you must use:

```
Hdf5ConvertibleType = 
Dict[str, Union[int, float, Batch, np.ndarray, torch.Tensor, object, 'Hdf5ConvertibleType']]
```

Or, if you want to avoid redundancy (since everything derives from `object`):

```
Hdf5ConvertibleType =  Dict[str, Any]
```

Unfortunately, you cannot do any better... Personally I would rather go for the first one, because it carries the intend, but it is a matter of taste.

Reply:
Thank you for your advice and for helping me with the code! I have implemented your suggestion.